p2panda-net/src/lib.rs

```diff
@@ -1,5 +1,140 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
+//! `p2panda-net` is a data-type-agnostic p2p networking layer offering robust, direct
+//! communication to any device, no matter where they are.
+//!
+//! It provides a stream-based API for higher layers: Applications subscribe to any "topic" they
+//! are interested in and `p2panda-net` will automatically discover similar peers and transport raw
+//! bytes between them.
+//!
+//! Additionally `p2panda-net` can be extended with custom sync protocols for all data types,
+//! allowing applications to "catch up on past data", eventually converging to the same state.
+//!
+//! ## Features
+//!
+//! Most of the lower-level networking of `p2panda-net` is made possible by the work of
+//! [iroh](https://github.com/n0-computer/iroh/) utilising well-established and known standards,
+//! like QUIC for transport, (self-certified) TLS for transport encryption, STUN for establishing
+//! direct connections between devices, Tailscale's DERP (Designated Encrypted Relay for Packets)
+//! for relay fallbacks, PlumTree and HyParView for broadcast-based gossip overlays.
+//!
+//! p2panda adds crucial functionality on top of iroh for peer-to-peer application development,
+//! without tying developers too close to any pre-defined data types and allowing plenty space for
+//! customisation:
+//!
+//! 1. Data of any kind can be exchanged efficiently via gossip broadcast ("live mode") or via sync
+//!    protocols between two peers ("catching up on past state")
+//! 2. Custom network-wide queries to express interest in certain data of applications
```

Comment:
Might be nice to mention that these queries are "subjective" as in any peer can define themselves what they are interested in.

Reply:
Good idea! I find it hard to bring that in in the main features section, but will try when where there's more space to explain topics